harc_game_web/apps/posts/views.py

```diff
@@ -0,0 +1,92 @@
+from django.shortcuts import render, redirect, get_object_or_404, reverse
+from django import forms
+from .models import Post
+from django.http import HttpResponseRedirect
+from django.utils import timezone
+from django.db.models import F
+
+class PostForm(forms.ModelForm):
+    class Meta:
+        model = Post
+        fields = ['pub_date_time', 'title', 'content', 'link1', 'link2', 'link3']
+  
+def ListActivePosts(request):
+	"""
+	Function to list all posts on home screen
+	"""
+	# Exclude future posts
+	posts = Post.objects.exclude(pub_date_time__gte = timezone.now())
+	return render(request, 'posts/list_active.html', {'posts': posts})
+	
+def ListAllPosts(request):
+	"""
+	Function to list all posts
+	"""
+	posts = Post.objects.all().order_by(F('pub_date_time').desc(nulls_last=True))
+	return render(request, 'posts/list.html', {'posts': posts})
+
+def ViewPost(request, slug):
+	"""
+	Function to list a single post
+	"""
+	post = Post.objects.get(slug=slug)
+	return render(request, 'posts/view.html', {'post': post})
+	
+def NewPost(request):
```

Comment:
This and other functions (edit, delete post) should be available only for superuser. Decorator `@staff_member_required` or other check should be used

Reply:
It is actually done in html templat. Will try @staff_member_required